### PR TITLE
Write new pending message format in PendingStateManager

### PR DESCRIPTION
**Pull request by kian-thompson**
In PendingStateManager.getLocalState(), write the new message format instead of the old format
See changes in https://github.com/microsoft/FluidFramework/pull/14462

[AB#3826](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3826)

Follow-up item for 2.0.0-internal.7.0.0: [AB#4763](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4763)

### CODE REVIEW

File: packages/runtime/container-runtime/src/pendingStateManager.ts
Comment (kian-thompson) [inline packages/runtime/container-runtime/src/pendingStateManager.ts:119]: @justus-camp Is it possible to have this `stashedState` be part of the content when submitting the message (i.e. in `maybeSubmitIdAllocationOp`)?
I want to avoid `parse` then `serialize` as much as possible just to check the type add another prop to the content string

Comment (justus-camp) [reply packages/runtime/container-runtime/src/pendingStateManager.ts:119]: I used localOpMetadata because we don't want to submit the `stashedState` over the wire but need it to rehydrate the IdCompressor for stashed ops. My initial prototype actually did what you're suggesting and deleted `stashedState` from the op before submission, but there's [a check](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/container-runtime/src/pendingStateManager.ts#L241C21-L241C21) in `PendingStateManager::processPendingLocalMessage` that makes doing it this way difficult and causes a `DataProcessingError`. I suppose we could ignore `IdAllocation` ops in that check but that's pretty unfortunate. 

It's worth noting that overall this stashed op implementation isn't great and we kind of "retro-fitted" the original compressor to do it this way.

Comment (kian-thompson) [reply packages/runtime/container-runtime/src/pendingStateManager.ts:119]: Is it possible to move the serialized state to a summary blob? I think it should, but I may be missing something. If so, we can leave logic in the `PendingStateManager` for back/forward compat while that goes on.

If not possible, I'll add some logic in `ContainerRuntime` to handle this conversion of content for the `PendingStateManager`. The PSM shouldn't know about specific types of ops if avoidable.

**Review by NicholasCouri — APPROVED** [review]
:shipit:

**Comment by sonalideshpandemsft** [comment]
I'll merge this PR after main/next integrate PR is merged: https://github.com/microsoft/FluidFramework/pull/16428

**Pull request merged**
